### PR TITLE
Remove gce presubmit using dockershim

### PR DESCRIPTION
**Pull request by matthyx**
relates to #24620

### CODE REVIEW

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.23.yaml [hunk 1]
```diff
@@ -1031,6 +1031,8 @@ presubmits:
       preset-pull-kubernetes-e2e-gce: "true"
       preset-service-account: "true"
     name: pull-kubernetes-e2e-gce-ubuntu
+    annotations:
```
Comment (SergeyKanzhelev) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.23.yaml:1034]: is this change intentional?

Comment (matthyx) [reply config/jobs/kubernetes/sig-release/release-branch-jobs/1.23.yaml:1034]: yes it's needed, because L216 in other file created the group... we did the same in https://github.com/kubernetes/test-infra/pull/24911

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Comment by matthyx** [comment]
Thanks @SergeyKanzhelev !
/assign @cheftako @jeremyrickard

**Comment by dims** [comment]
@matthyx please rebase and i'll approve

**Comment by matthyx** [comment]
@dims  done, thanks!
